conjure-codegen/src/errors.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use proc_macro2::TokenStream;
+use quote::quote;
+
+use crate::context::Context;
+use crate::objects;
+use crate::types::{ErrorDefinition, ObjectDefinition};
+
+pub fn generate(ctx: &Context, def: &ErrorDefinition) -> TokenStream {
+    let object = ObjectDefinition::builder()
+        .type_name(def.error_name().clone())
+        .fields(def.safe_args().iter().chain(def.unsafe_args()).cloned())
```

Comment:
spec seems iffy on whether you can share keys between safe args and unsafe args

Reply:
ok, there doesn't seem to be any validation on this and I was able to get conjure-java to produce a non-compiling error 

```java
    public static ServiceException myError(
            Throwable cause, int param, List<String> param) {
        return new ServiceException(
                MY_ERROR,
                cause,
                SafeArg.of("param", param),
                UnsafeArg.of("param", param));
    }
```

Reply:
(so probably not a thing to worry about here, but I'll put a bug on the conjure IR project)

Reply:
https://github.com/palantir/conjure/issues/285